app/src/main/java/com/example/tkey_android/FirstFragment.java

```diff
@@ -158,6 +164,83 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
                 throw new RuntimeException(e);
             }
         });
+
+        binding.requestNewShare.setOnClickListener(view1 -> {
+            requireActivity().runOnUiThread(() -> {
+                String userAgent = new WebView(getContext()).getSettings().getUserAgentString();
+                try {
+                    SharetransferModule.requestNewShare(activity.appKey, userAgent, activity.appKey.getShareIndexes().get(0), (result) -> {
```

Comment:
I don't think you want to do this for available indexes `activity.appKey.getShareIndexes().get(0)`
```suggestion
                    SharetransferModule.requestNewShare(activity.appKey, userAgent, activity.appKey.getShareIndexes().get(0), (result) -> {
```